### ISSUE TITLE
loss nan

### ISSUE THREAD

**Issue by liyangliu** [jwyang/faster-rcnn.pytorch#39]
Hello @jwyang, I downloaded your pytorch faster rcnn yesterday, only change the coco 2014 dataset path to my local one and trained exactly the same setting as you (large image scale, lr = 0.01, 2 images per gpu and 8 gpus, res101, using caffe pretrained models given by you) but got **NAN** loss after a few iteration. Have you come across this problem? The loss will **not** be NAN if I set class_agnostic=True. Can you please help me a little? Thanks.

**Comment by jwyang** [comment]
Hi, @liyangliu , could you share your training command and training log?

I did not encounter this problem.

**Comment by liyangliu** [comment]
python trainval_net.py \
    --dataset coco \
    --net res101 \
    --save_dir=exps/baseline/models \
    --cuda \
    --mGPUs \
    --bs 16 \
    --nw 8 \
    --epochs 10 \
    --ls \
    --lr 0.01 \
    --lr_decay_step 4 \
    &> logs/baseline.log &

**Comment by gyxoned** [comment]
Hi, I met this problem several times before. But it was solved when I run it again without any setting changes.

**Comment by jwyang** [comment]
@liyangliu  I think we might have slightly different initializations. If you encounter this again, one way is to clamp the gradient for res101 as well by comment this [line](https://github.com/jwyang/faster-rcnn.pytorch/blob/df7685772c3ab379ee2ca3dab35809b49e877629/trainval_net.py#L331).

@gyxoned, have you successfully trained the model and get similar performance as reported in our tables?

**Comment by jwyang** [comment]
@gyxoned sounds great!

**Comment by shenshanlaoma** [comment]
modify these 4 lines！delete “- 1”！ 
            x1 = float(bbox.find('xmin').text) - 1
            y1 = float(bbox.find('ymin').text) - 1
            x2 = float(bbox.find('xmax').text) - 1
            y2 = float(bbox.find('ymax').text) - 1[https://github.com/jwyang/faster-rcnn.pytorch/blob/42b92b1fb2df0ac4a3669bf47a7b589d466a5ea8/lib/datasets/pascal_voc.py#L234](url)
according to [http://caffecn.cn/?/question/1055](url) [https://stackoverflow.com/questions/38513739/warning-during-py-faster-rcnn-training-on-custom-datasets](url)

**Comment by TomHeaven** [comment]
In my case, clamp the gradient for res101 is the correct solution of nan loss.
>> @liyangliu I think we might have slightly different initializations. If you encounter this again, one way is to clamp the gradient for res101 as well by comment this [line](https://github.com/jwyang/faster-rcnn.pytorch/blob/df7685772c3ab379ee2ca3dab35809b49e877629/trainval_net.py#L331).

**Comment by tangbohu** [comment]
> modify these 4 lines！delete “- 1”！
> x1 = float(bbox.find('xmin').text) - 1
> y1 = float(bbox.find('ymin').text) - 1
> x2 = float(bbox.find('xmax').text) - 1
> y2 = float(bbox.find('ymax').text) - 1
> 
> https://github.com/jwyang/faster-rcnn.pytorch/blob/42b92b1fb2df0ac4a3669bf47a7b589d466a5ea8/lib/datasets/pascal_voc.py#L234
> 
> 
> according to http://caffecn.cn/?/question/1055 https://stackoverflow.com/questions/38513739/warning-during-py-faster-rcnn-training-on-custom-datasets

should we delete "-1" on the experiments on VOC?
